wp-admin/js/widgets/media-image-widget.js

```diff
@@ -106,7 +106,7 @@
 					image_title: '',
 					link_classes: '',
 					link_rel: '',
-					link_url: attachment.linkUrl ? attachment.linkUrl : attachment.link,
+					link_url: attachment.linkUrl,
```

Comment:
`linkUrl` is only set when it's a custom link. When it's linked to the image, it doesn't pass the image url again.
So it looks like we need to be more verbose here and check for what's in `link` and populate `link_url` accordingly.